client/state/ui/guided-tours/contexts.js

```diff
@@ -19,15 +20,20 @@ export const isPreviewNotShowing = state =>
 export const isPreviewShowing = state =>
 	isPreviewShowingSelector( state );
 
+const timeSinceUserRegistration = state => {
+	const user = getCurrentUser( state );
+	return user ? ( Date.now() - Date.parse( user.date ) ) : false;
+};
+
 const WEEK_IN_MILLISECONDS = 7 * 1000 * 3600 * 24;
 export const isNewUser = state => {
-	const user = getCurrentUser( state );
-	if ( ! user ) {
-		return false;
-	}
+	const userAge = timeSinceUserRegistration( state );
+	return ! isNaN( userAge ) ? userAge <= WEEK_IN_MILLISECONDS : false;
```

Comment:
Looking at `timeSinceUserRegistration`, the return type is `number | false | NaN`, but we aren't checking for `false` here. Notably, `isNaN( false ) === false`, and `false <= number === true`, so the logic here seems flawed.